app/controllers/media_controller.rb

```diff
@@ -0,0 +1,101 @@
+class MediaController < ApplicationController
+  before_action :set_type
+  before_action :creator, only: [:new, :edit, :create, :update]
+
+  def index
+    @items = type_class.order(votes: :desc)
+
+    render "index"
+  end
+
+  def show
+    @item = type_class.find(params[:id])
```

Comment:
You use this code snippet a few times. Perhaps you could DRY things up by extracting it into a before action.